core/tauri-utils/src/config.rs

```diff
@@ -1920,28 +1773,30 @@ pub enum HookCommand {
 pub struct BuildConfig {
   /// The binary used to build and run the application.
   pub runner: Option<String>,
-  /// The path to the application assets or URL to load in development.
+  /// The URL to load in development.
   ///
-  /// This is usually an URL to a dev server, which serves your application assets
-  /// with live reloading. Most modern JavaScript bundlers provides a way to start a dev server by default.
+  /// This is usually an URL to a dev server, which serves your application assets with hot-reload and HMR.
+  /// Most modern JavaScript bundlers like [vite](https://vitejs.dev/guide/) provides a way to start a dev server by default.
   ///
-  /// See [vite](https://vitejs.dev/guide/), [Webpack DevServer](https://webpack.js.org/configuration/dev-server/) and [sirv](https://github.com/lukeed/sirv)
-  /// for examples on how to set up a dev server.
-  #[serde(alias = "dev-path")]
-  pub dev_path: Option<AppUrl>,
-  /// The path to the application assets or URL to load in production.
+  /// If you don't have a dev server or don't want to use one, ignore this option and use [`frontendDist`](BuildConfig::frontend_dist)
+  /// and point to a web assets directory, and Tauri CLI will run its built-in dev server and provide a simple hot-reload experience.
+  #[serde(alias = "dev-url")]
+  pub dev_url: Option<Url>,
+  /// The path to the application assets (usually the `dist` folder of your javascript bundler)
+  /// or a URL that could be either a custom protocol registered in the tauri app (for example: `myprotocol://`)
+  /// or a remote URL (for example: `https://site.com/app`).
   ///
   /// When a path relative to the configuration file is provided,
   /// it is read recursively and all files are embedded in the application binary.
-  /// Tauri then looks for an `index.html` file unless you provide a custom window URL.
+  /// Tauri then looks for an `index.html` and serves it as the default entry point for your application.
   ///
   /// You can also provide a list of paths to be embedded, which allows granular control over what files are added to the binary.
   /// In this case, all files are added to the root and you must reference it that way in your HTML files.
   ///
-  /// When an URL is provided, the application won't have bundled assets
+  /// When a URL is provided, the application won't have bundled assets
   /// and the application will load that URL by default.
-  #[serde(alias = "dist-dir")]
-  pub dist_dir: Option<AppUrl>,
+  #[serde(alias = "frontend-dist")]
+  pub frontend_dist: Option<AppUrl>,
```

Comment:
this option seems overloaded IMO and `frontend_dist` doesn't really convey all possible options, do we want to split this? maybe into:
- `frontend_dist`
- `frontend_files`
- `frontend_url`

Reply:
the problem is they are mutually exclusive...

Reply:
what if we turn it into an enum?
`production: { url: string } | { dist: pathbuf } | { files: vec<pathbuf> }`

Reply:
another problem with frontend_url is that it would be weird next to dev_url

Reply:
spoke with @FabianLars and we agree it's better to keep it as is...